share/module_check_a_mundo.F

```diff
@@ -777,6 +777,21 @@ END FUNCTION bep_bem_nbui_max
          END IF
       ENDDO
 
+!-----------------------------------------------------------------------
+! Check that all values of gwd_opt are the same for all domains
+!-----------------------------------------------------------------------
+
+      DO i = 2, model_config_rec % max_dom
+         IF ( .NOT. model_config_rec % grid_allowed(i) ) CYCLE
+         IF ( ( model_config_rec % gwd_opt(i) .NE. model_config_rec % gwd_opt(1) ) .AND. &
+              ( model_config_rec % gwd_opt(i) .NE. 0                             ) ) THEN 
+            wrf_err_message = '--- ERROR: gwd_opt must be equal for all domains (or = zero)'
+            CALL wrf_message ( wrf_err_message )
+            wrf_err_message = '--- Fix gwd_opt in namelist.input '
+         CALL wrf_debug ( 0, TRIM( wrf_err_message ) )
+         count_fatal_error = count_fatal_error + 1
+         END IF
+      ENDDO
```

Comment:
@mdtoy Is this needed for gwd_opt = 3? We have changed this to allow gwd_opt = 1 to be off for smaller grid distances. If you would like to enforce this for gwd_opt = 3, please make it explicit.

Reply:
My intention was to have gwd_opt=3 treated exactly like gwd_opt=1, that is, to allow the scheme to be off for smaller grid distances.  These lines also prevent different non-zero gwd_opts from being used, i.e., only one gwd_opt can be used across the domains.  Can you clarify what you mean by making it explicit?

Reply:
@mdtoy I guess I'm reading the code wrong. As long as it allows a domain to use 0 option, then this should be good. It is also good to check if the option is the same for each domain. Sorry.

Reply:
I agree that this is the correct logic, to allow finer-scale domains to turn off GWD